src/cli/self_update/windows.rs

```diff
@@ -402,26 +411,33 @@ pub fn delete_rustup_and_cargo_home() -> Result<()> {
 
 #[cfg(test)]
 mod tests {
+    use std::ffi::OsString;
+    use std::os::windows::ffi::OsStrExt;
+
     use winreg::enums::{RegType, HKEY_CURRENT_USER, KEY_READ, KEY_WRITE};
     use winreg::{RegKey, RegValue};
 
     use crate::currentprocess;
     use crate::test::with_saved_path;
     use crate::utils::utils;
 
+    fn wide(str: &str) -> Vec<u16> {
+        OsString::from(str).encode_wide().collect()
+    }
+
     #[test]
     fn windows_install_does_not_add_path_twice() {
         assert_eq!(
             None,
             super::_add_to_path(
-                r"c:\users\example\.cargo\bin;foo",
-                r"c:\users\example\.cargo\bin".into()
+                wide(r"c:\users\example\.cargo\bin;foo"),
+                wide(r"c:\users\example\.cargo\bin")
             )
         );
     }
 
     #[test]
-    fn windows_doesnt_mess_with_a_non_unicode_path() {
+    fn windows_doesnt_mess_with_a_non_string_path() {
```

Comment:
this is still a non-unicode path, and we still want it not-messed with, so I don't think the title needs altering, does it? Perhaps you want to change it to "non_unicode_paths_are_handled_safely" or something